pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -1469,3 +1472,58 @@ type IonosSDConfig struct {
 	// +optional
 	OAuth2 *v1.OAuth2 `json:"oauth2,omitempty"`
 }
+
+// StackitSDConfig configurations enables service discovery for STACKIT.
+// See https://prometheus.io/docs/prometheus/3.5/configuration/configuration/#stackit_sd_config
+type StackitSDConfig struct {
+	// The STACKIT Project ID.
+	// +kubebuilder:validation:MinLength:=1
+	// +required
+	Project string `json:"project"`
+	// The STACKIT region.
+	// +optional
+	Region *string `json:"region,omitempty"`
+	// The STACKIT endpoint.
+	// +kubebuilder:validation:Pattern:="^http(s)?://.+$"
+	// +optional
+	Endpoint *string `json:"endpoint"`
+	// Default port to scrape metrics from.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=65535
+	// +optional
+	Port *int32 `json:"port,omitempty"`
+	// Raw private key string used for authenticating a service account.
+	// +optional
+	PrivateKey *string `json:"privateKey"`
+	// Path to a file containing the raw private key string.
+	// +optional
+	PrivateKeyPath *string `json:"privateKeyPath"`
```

Comment:
We would prefer to not have fields that point to filepaths in the config resources. 
```suggestion
```